tests/rptest/scale_tests/many_partitions_test.py

```diff
@@ -989,3 +997,530 @@ def progress_check():
                     f"Expected throughput {expect_mbps:.2f}MiB/s, got throughput {actual_mbps:.2f}MiB/s"
                 )
                 raise
+
+    def _try_parse_json(self, node, jsondata):
+        try:
+            return json.loads(jsondata)
+        except ValueError:
+            self.logger.debug(
+                f"{str(node.account)}: Could not parse as json: {str(jsondata)}"
+            )
+            return None
+
+    def _write_and_random_read_many_topics(self, num_topics, topic_names):
```

Comment:
nit: It would be nice to split these utility methods into separate classes to may them reusable in other tests. Not something we should focus on doing right now though.